### PR TITLE
Admin CRUD operations over Users

### PR DESCRIPTION
**Pull request by BDesselle**
# Description

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: auth/authRoute.js [hunk 1]
```diff
@@ -119,7 +121,7 @@ router.post("/register", (req, res) => {
       .then(user => {
         const token = generateToken(user);
         res.status(201).json({
```
Comment (agordon1123) [inline auth/authRoute.js]: We could `delete user.password` and then send back the user object without that key value pair.

File: middleware/generateToken.js [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
-    const options = {
-        expiresIn: '1d'
-    }
-
-    return jwt.sign(payload, secrets.jwtSecret, options)
+function generateToken(user) {
+  //* Changing the secret used based off of the user's role
+  try {
+    user.role === "admin"
+      ? ((secret = secrets.adminSecret), console.log("Admin Secret Used"))
```
Comment (agordon1123) [inline middleware/generateToken.js]: Can we remove the console.logs? Has it been tested?

File: middleware/generateToken.js [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
+      : user.role === "manager"
+      ? ((secret = secrets.managerSecret), console.log("Manager Secret Used"))
+      : user.role === "student"
+      ? ((secret = secrets.studentSecret), console.log("Student Secret Used"))
+      : res.status(400).json({
+          message: "Role Identification Not Found"
+        });
```
Comment (armandoroman1016) [inline middleware/generateToken.js]: where is the secret variable being initialized ?

File: middleware/isAdmin.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  const token = req.headers.authorization;
+
+  token
+    ? jwt.verify(token, secrets.adminSecret, (err, decodedToken) => {
+        err
+          ? res.status(401).json({ message: "Invalid Token" })
+          : decodedToken.role === "admin"
+          ? next()
+          : res.status(401).json({ message: "Invalid Permissions" });
+      })
```
Comment (armandoroman1016) [inline middleware/isAdmin.js]: do we cant to hard code the comparison here or set that to an .env variable ?

File: auth/authRoute.js (hunk 1 above)
Comment (zoverlvx) [reply auth/authRoute.js]: I've already got that set up in my PR. No worries to make any changes here @BDesselle

File: config/secrets.js [hunk 5]
```diff
@@ -2,5 +2,8 @@ module.exports = {
   jwtSecret:
     process.env.JWT_SECRET ||
     "4*Zey0zCfzeC@m2qmG&8@FHOVas3cPf$l&hr3m5mAzIa3pkxUPALgsi5kv7j9oYQ21J66aVIgE%6aj%SHEykGtY1RFmTJ5&*^Ukz",
+  adminSecret: process.env.JWT_ADMIN,
+  managerSecret: process.env.JWT_MANAGER,
+  studentSecret: process.env.JWT_STUDENT,
```
Comment (danbergelt) [inline config/secrets.js]: A reminder that we need to add env vars into the deployed application (through AWS)

**Comment by zoverlvx** [comment]
Make PR to merge `Roles` into `development`

**Pull request closed without merging**